### PR TITLE
Add projects view 

### PR DESCRIPTION
**Pull request by kresimir-coko**
Hey @bryceosterhaus I've added the projects view using the Cards components, but I didn't manage to get a single view working dynamically. If you give me a pointer I'm sure I can get it working, or given enough time I can crack the case.

## Some notes/ideas:
1. If more than 1 projects show the `Projects` view, otherwise default to the `Jobs` view.
2. Project is a parent of all its jobs, which means that every job will need a `project` property in the API
3. Project will probably need a `description` prop in the API most likely
4. Will the project want an image? Or a placeholder image of the service (github, gitlabs...), or some other indicator of the service used, as I assume not all projects will be hosted on the exact same service

### CODE REVIEW

**Comment by bryceosterhaus** [comment]
> Will the project want an image? Or a placeholder image of the service (github, gitlabs...), or some other indicator of the service used, as I assume not all projects will be hosted on the exact same service

I wouldn't worry about that right now. If i ever becomes more of a sass product, they can add that later but right now the focus is on running it locally only.

Not sure what you mean by "single view", but as for the routes, I think you need to nest our existing pages under a `[projectId]` folder. That would make it so we have the following routes.
| File structure | Web Route |
|----|----|
| `projects/index.tsx` | `/projects` | 
| `projects/[projectId]/index.tsx` | `/projects/{PROJECT_ID}` | 
| `projects/[projectId]/[jobs]/index.tsx` | `/projects/{PROJECT_ID}/jobs` | 
| `projects/[projectId]/[jobs]/[jobId].tsx` | `/projects/{PROJECT_ID}/jobs/{JOB_ID}` | 

You can learn more about dynamic pages [here](https://nextjs.org/docs/basic-features/pages#pages-with-dynamic-routes), and remember they work for both  files and directories.


And it is likely we will start with only a single project, but this at least gives lug bot flexibility to add more in the future.

File: pages/jobs/index.tsx [hunk 1]
```diff
@@ -270,11 +270,11 @@ export async function getServerSideProps(context) {
 		(res) => res.json()
 	);
 
-	const project = await fetch(
-		`${getAPIOrigin(context.req)}/api/project`
+	const projects = await fetch(
+		`${getAPIOrigin(context.req)}/api/projects`
```
Comment (bryceosterhaus) [inline pages/jobs/index.tsx]: When you nest this file under a dynamic projects directory, you will be able to get the id from the url via
```js
const {projectId} = context.req.query
```

**Comment by kresimir-coko** [comment]
> Not sure what you mean by "single view"

I meant single project view, my bad. Like `localhost:3000/projects/[projectId]`.

**Comment by kresimir-coko** [comment]
I got it working! A couple problems I couldn't fix, ran out of time:
- `projects` is under `jobs` in the dummy date API... `jobs: {runningJobs: {}, pendingJobs: {}, projects: {}...}`
- each job needs a `project`  prop so that we can fetch only jobs related to that project when we go to `projects/{project}/jobs`

**Comment by bryceosterhaus** [comment]
looks good! I added a few updates in this pr, https://github.com/liferay-frontend/lug-bot-app/pull/22, take a look at my commits and see what I changed. Nothing major, I just wanted to make sure the pages were redirecting properly and the API also using the nested data structure of projects

**Pull request closed without merging**